transport/src/main/java/io/netty/channel/AbstractChannelHandlerContext.java

```diff
@@ -872,12 +893,94 @@ private AbstractWriteTask(Recycler.Handle handle) {
             super(handle);
         }
 
+        private static int estimateSize(Object o, Map<Class<?>, Integer> classSizes) {
+            if (o == null) {
+                return 8;
+            }
+
+            int answer = 8 + estimateSize(o.getClass(), classSizes, null);
+
+            if (o instanceof ByteBuf) {
+                answer += ((ByteBuf) o).readableBytes();
+            } else if (o instanceof ByteBufHolder) {
+                answer += ((ByteBufHolder) o).content().readableBytes();
+            } else if (o instanceof FileRegion) {
+                answer += 0;
+            } else if (o instanceof byte[]) {
+                answer += ((byte[]) o).length;
+            } else if (o instanceof ByteBuffer) {
+                answer += ((ByteBuffer) o).remaining();
+            } else if (o instanceof CharSequence) {
+                answer += ((CharSequence) o).length() << 1;
+            } else if (o instanceof Iterable<?>) {
+                for (Object m : (Iterable<?>) o) {
+                    answer += estimateSize(m, classSizes);
+                }
+            }
+
+            return align(answer);
+        }
+
+        private static int estimateSize(Class<?> clazz, Map<Class<?>, Integer> classSizes,
```

Comment:
Wondering if all of this reflection is going to prove to be a performance issue.  I'm guessing that the expectation here is that this is only done once per class type, so it doesn't matter?